h2o-genmodel/src/main/java/hex/genmodel/algos/ensemble/StackedEnsembleMojoModel.java

```diff
@@ -2,43 +2,90 @@
 
 import hex.genmodel.MojoModel;
 
+import java.util.Arrays;
+
 public class StackedEnsembleMojoModel extends MojoModel {
 
     MojoModel _metaLearner; //Currently only a GLM. May change to be DRF, GBM, XGBoost, or DL in the future
-    MojoModel[] _baseModels; //An array of base models
+    StackedEnsembleMojoSubModel[] _baseModels; //An array of base models
     int _baseModelNum; //Number of base models
+    int _numMappings; // Number of mappings against referential column order
+    double[][] _precalculatedMappings;
```

Comment:
MOJO needs to be stateless, this is not threadsafe

Reply:
To be fair not all mojos are threadsafe, Deep Learning MOJO isn't but there is a good reason for it. I would go for stateless unless needed.